src/pages/plans/Plans.tsx

```diff
@@ -1,96 +1,241 @@
+import CreatePlanForm from "./CreatePlanForm/CreatePlanForm";
+import {
+  useCreatePlanMutation,
+  useUpdatePlanPriceMutation,
+  useDeletePlanMutation,
+} from "../../app/services/PlanService";
+import { useGetUserPlansQuery } from "../../app/services/UserService";
+import { skipToken } from "@reduxjs/toolkit/query/react";
+import useAuthentication from "../../hooks/useAuthentication";
+import EditPlan from "./EditPlan/EditPlan";
 
 function Plans() {
+  const formRef = useRef<any>(null);
+  const [filter, setFilter] = useState<string>("");
+  const [planList, setPlanList] = useState<PlanResponse[]>([]);
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [showCreateModal, setShowCreateModal] = useState(false);
+  const [showEditModal, setShowEditModal] = useState(false);
+  const [selectedPlanId, setSelectedPlanId] = useState<string | null>(null);
+  const [selectedPlan, setSelectedPlan] = useState<PlanResponse | null>(null);
+  const [editPlan] = useUpdatePlanPriceMutation();
+  const [deletePlan] = useDeletePlanMutation();
+  const [createPlan] = useCreatePlanMutation();
   const { userId } = useAuthentication();
-  const { data: prices } = useGetUserPricesQuery(userId!);
-  const [newAmount, setNewAmount] = useState<number>(0);
-  const [selectedPriceId, setSelectedPriceId] = useState("");
-  const [updatePrice, { isLoading: isUpdating }] = useUpdatePriceMutation();
-  const selectedPrice = prices?.find((p) => p.id === selectedPriceId);
-  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const rawValue = e.target.value.replace(/[^\d]/g, "");
-    setNewAmount(Number(rawValue) / 100);
-  };
-
-  const handleSubmit = (e: React.FormEvent) => {
-    e.preventDefault();
-    if (!selectedPriceId) {
-      toast.error("Por favor, selecciona un plan.");
-      return;
+  const {
+    data: plans,
+    isLoading,
+    isError,
+  } = useGetUserPlansQuery(userId ? userId : skipToken);
+  useMemo(() => {
+    if (plans) {
+      setPlanList(plans);
     }
+  }, [plans]);
+
+  //Filtro de busqueda por nombre desde el frontend, cuando
+  //este el backend se tiene que modificar:
+  const filteredPlans = useMemo(() => {
+    if (!filter) return planList;
 
-    if (!newAmount || isNaN(Number(newAmount))) {
-      toast.error("Ingresa un monto válido.");
-      return;
+    return planList.filter((plan) =>
+      plan.name.toLowerCase().includes(filter.toLowerCase())
+    );
+  }, [planList, filter]);
+
+  const handleClearFilters = () => {
+    setFilter("");
+  };
+  const handleMutation = async (
+    action: () => Promise<any>,
+    onSuccess: () => void,
+    successMessage: string,
+    errorMessage: string
+  ) => {
+    try {
+      await action();
+      toast.success(successMessage);
+      onSuccess();
+    } catch (error: any) {
+      toast.error(errorMessage);
     }
+  };
+  //CREAR PLAN
+  const handleCreateModal = async () => {
```

Comment:
En vez de tener un comentario "Crear plan" lo que haria es cambiarle el nombre al metodo para que sea mas representativo


```suggestion
  const handleCreatePlan = async () => {
```